File: CS50x - 2025/sentimental-mario-less/mario.py
from cs50 import get_int
## ensure valid input; prompt user if h is <1 or >8
while True:
    h = get_int('height: ')
    if h < 1 or h > 8:
        h = get_int('height: ')
    if h >= 1 or h <= 8:
        break
## iterate through h
for i in range(h):
    print((h - 1 - i) * " ", end="")
    print((i + 1) * "#")

File: CS50x - 2025/mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{

    int height;

    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    for (int row = 0; row < height; row++)
    {
        for (int column = 0; column < height; column++)
        {
            if (row + column >= height - 1)
                printf("#");

            else
                printf(" ");
        }
        printf("\n");
    }
}
